components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java

```diff
@@ -154,6 +153,7 @@ public OAuthConsumerAppDTO getOAuthApplicationData(String consumerKey) throws Id
         try {
             OAuthAppDO app = dao.getAppInformation(consumerKey);
             if (app != null) {
+                AppInfoCache.getInstance().addToCache(consumerKey, app);
```

Comment:
Is this method being used throughout to retrieve app from db always.

Reply:
Yes. It seems so.
In fact, none of the App information retrieval within this class make use of the cache. 

Ideally, we should have a cache backed DAO. But for the moment let me improve the code to make use of app info cache before trying to retrieve from DB

Reply:
+1